src/libs/kata-types/src/annotations/mod.rs

```diff
@@ -736,14 +736,9 @@ impl Annotation {
                             return Err(bool_err);
                         }
                     },
-                    KATA_ANNO_CFG_HYPERVISOR_HUGE_PAGES => match self.get_value::<bool>(key) {
-                        Ok(r) => {
-                            hv.memory_info.enable_hugepages = r.unwrap_or_default();
-                        }
-                        Err(_e) => {
-                            return Err(bool_err);
-                        }
-                    },
+                    KATA_ANNO_CFG_HYPERVISOR_HUGE_PAGES => {
+                        hv.memory_info.hugepages = value.to_string();
```

Comment:
The local style is to validate annotations here, see for example line 681:

```
                        hv.machine_info.validate_entropy_source(value)?;
```

I believe that this should be done independently of a possibly later hypervisor-specific check as you do for Dragonball.

Reply:
https://github.com/kata-containers/kata-containers/blob/7fdaab49bcce816cd0f928677cf67dcea9548ec3/src/libs/kata-types/src/config/hypervisor/mod.rs#L600-L612

Reply:
That's a good place to start I think